crud/select/plan.lua

```diff
@@ -235,6 +277,17 @@ function select_plan.new(space, conditions, opts)
         sharding_key = extract_sharding_key_from_scan_value(scan_value, scan_index, sharding_index)
     end
 
+    if sharding_key == nil then
+        -- Ignore possible errors to preserve old behavior
+        -- since here it affects only extracting sharding_key from
+        -- conditions and it should not be critical to select call.
+        local ddl_sharding_key = sharding_key_module.fetch_on_router(space_name)
+
+        sharding_key = extract_sharding_key_from_conditions(
+            conditions, ddl_sharding_key, space_indexes, space_field_positions
+        )
```

Comment:
Does not it supersedes `extract_sharding_key_from_scan_value()`? I mean, handles all its cases and even more.

Reply:
Yeah, you were right. New reworked version changed a behavior in plan unit test. I think it is as expected and an improvement rather than drawback, but maybe it should be double-checked. Using `extract_sharding_key_from_conditions` may introduce minor performance drop for a case when sharding value is exactly scan value since all cases treated equally and `scan_value` is not used here.

Reply:
That's two checks: whether the key is not partial and whether scan index is sharding index. We can add it back as fast path.

Reply:
Added it

Reply:
Nice.